cdap-system-app-api/src/main/java/io/cdap/cdap/api/service/worker/RemoteExecutionException.java

```diff
@@ -42,13 +42,19 @@ public RemoteTaskException getCause() {
    * @return An exception which retains the local stacktrace.
    */
   public static RemoteExecutionException fromBasicThrowable(BasicThrowable basicThrowable) {
+    RemoteTaskException remoteTaskException = getRemoteTaskException(basicThrowable);
+    // Wrap the remote exception as the cause so that we retain the local stacktrace of the exception.
+    return new RemoteExecutionException(remoteTaskException);
+  }
+
+  private static RemoteTaskException getRemoteTaskException(BasicThrowable basicThrowable) {
```

Comment:
nit: Let's call this `recursiveWrapBasicThrowable` for clarity, thanks!

Reply:
Sounds good, updated